web/themes/custom/jcc_components/includes/views.inc

```diff
@@ -431,3 +432,129 @@ function jcc_components_views_pre_render(ViewExecutable $view){
       break;
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function jcc_components_preprocess_views_view__feeds_directory__page_1(&$variables) {
+  if (empty($variables['rows'])) {
+    return;
+  }
+  $feeds_types = [];
+  $feeds_topics = [];
+  foreach ($variables['rows'][0]['#rows'] as $row) {
+    $term = $row['#row']->_entity;
+    $term_name = $term->name->value;
+    $tid = $term->tid->value;
+    $vid = Term::load($tid)->bundle();
+    $feed_details = [
+      'icon' => [
+        'name' => 'rss2',
+        'title' => 'rss2',
+      ],
+      'name' => $term_name,
+      'url' => '/cats/' . $tid,
+    ];
+
+    switch ($vid) {
+      case 'news_type':
+        array_push($feeds_topics, $feed_details);
+        break;
+
+      default:
+        array_push($feeds_types, $feed_details);
+    }
+  }
+
+  $feeds_types = $feeds_topics;
+  foreach ($feeds_types as $key => $type) {
+    if ($type['name'] != 'News Release') {
+      unset($feeds_types[$key]);
+    }
+  }
+  $feeds_types = array_values($feeds_types);
+  $feeds_types[0]['name'] = 'News & Events';
+  $feeds_types[0]['url'] = '/news-events';
+  $feeds = [
+    [
+      'feed_type' => 'News Feed',
+      'feed_items' => $feeds_types,
+    ],
+    [
+      'feed_type' => 'Categories',
+      'feed_items' => $feeds_topics,
+    ],
+  ];
+
+  $variables['feeds'] = $feeds;
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function jcc_components_preprocess_views_view_row_rss(&$variables) {
+  // Extending RSS content.
+  $item = $variables['row'];
+  $node = Node::load($item->nid);
+
+  if (isset($node)) {
+    $content_type = $node->type->entity->label();
+    $extend_rss = FALSE;
+    $tid = '';
+    switch ($content_type) {
+      case 'News':
+        $extend_rss = TRUE;
+        $tid = $node->get('field_news_type')->target_id;
+        break;
+
+      case 'Event':
```

Comment:
You are checking for Event content here but not in the directory view. Was this intentional?

Reply:
yes it is intensional